packages/block-library/src/navigation/edit/navigation-menu-selector.js

```diff
@@ -98,26 +93,32 @@ export default function NavigationMenuSelector( {
 				<>
 					{ showNavigationMenus && hasNavigationMenus && (
 						<MenuGroup label={ __( 'Menus' ) }>
-							{ navigationMenusOmitCurrent.map( ( menu ) => {
-								const label = decodeEntities(
-									menu.title.rendered
-								);
-								return (
-									<MenuItem
-										onClick={ () => {
-											onClose();
-											onSelect( menu );
-										} }
-										key={ menu.id }
-										aria-label={ sprintf(
-											actionLabel,
-											label
-										) }
-									>
-										{ label }
-									</MenuItem>
-								);
-							} ) }
+							<MenuItemsChoice
+								value={ currentMenuId }
+								onSelect={ ( selectedId ) => {
```

Comment:
nit: useCallback would be nice for that to avoid passing in a different object each time